library/core/src/alloc/global.rs

```diff
@@ -125,8 +125,8 @@ pub unsafe trait GlobalAlloc {
     ///
     /// # Safety
     ///
-    /// This function is unsafe because undefined behavior can result
-    /// if the caller does not ensure that `layout` has non-zero size.
+    /// The caller has to ensure that `layout` has non-zero size. Zero sized
+    /// `layout` can result in undefined behaviour.
```

Comment:
The cleanup is nice, but I don't think specifying "the caller" adds much here. Maybe something like:

```rust
    /// `layout` must have non-zero size. Attempting to allocate for a zero-sized
    /// `layout` may result in undefined behavior.
```

(optional nit)